src/main/resources/react4xp/_entries/CookieBanner.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react'
+import { Button } from '@statisticsnorway/ssb-component-library'
+import { blockOptionalCookies } from '../utils/cookieBannerUtils'
+
+const MAX_AGE = 7776000
+
+function CookieBanner() {
+  const [visible, setVisible] = useState(false)
+
+  useEffect(() => {
+    const hasConsent = document.cookie.includes('cookie-consent=')
+    if (!hasConsent) {
+      blockOptionalCookies()
+      setVisible(true)
+    }
+  }, [])
+
+  const setConsent = (type: 'all' | 'necessary') => {
+    if (type === 'necessary') {
+      blockOptionalCookies()
+    }
+
+    document.cookie = `cookie-consent=${type}; path=/; max-age=${MAX_AGE}; SameSite=Lax`
+
+    setVisible(false)
+  }
+
+  if (!visible) return null
+
+  return (
+    <div className='cookie-banner'>
+      <div className='cookie-banner-content'>
+        <h3 className='cookie-banner-title'>Vi bruker informasjonskapsler</h3>
```

Comment:
Check if it's possible to use more ssb components so we have less css code. E.g. could we use Title here instead?